subsys/bluetooth/controller/ll_sw/ull_llcp_hci.c

```diff
@@ -0,0 +1,999 @@
+			 *  conn->llcp_conn_param.state = cmd;
+			 *  conn->llcp_conn_param.cmd = 1U;
+			 *  conn->llcp_conn_param.req++;
+			 */
+		}
+
+#else /* !CONFIG_BT_CTLR_CONN_PARAM_REQ */
+		/* CPR feature not supported */
+		return BT_HCI_ERR_CMD_DISALLOWED;
+#endif /* !CONFIG_BT_CTLR_CONN_PARAM_REQ */
+	}
+
+	if (!cmd) {
+		/* replace with call to correct ull_cp_conn procedure */
+		return BT_HCI_ERR_UNKNOWN_CMD;
+	} else {
+		return BT_HCI_ERR_UNKNOWN_CMD;
+	}
+}
+
+
+
+u8_t ll_chm_get(u16_t handle, u8_t *chm)
+{
+	struct ull_cp_conn *conn;
+
+	conn = ll_connected_get(handle);
+	if (!conn) {
+		return BT_HCI_ERR_UNKNOWN_CONN_ID;
+	}
+
+	/*
+	 * EGON TODO: copy channelmap in connection structure, as:
+	 *	memcpy(chm, conn->lll.data_chan_map,
+	 *	       sizeof(conn->lll.data_chan_map));
+	 */
+	return BT_HCI_ERR_UNKNOWN_CMD;
+}
+
+
+
+#if defined(CONFIG_BT_CTLR_CONN_RSSI)
+u8_t ll_rssi_get(u16_t handle, u8_t *rssi)
+{
+	struct ull_cp_conn *conn;
+
+	conn = ll_connected_get(handle);
+	if (!conn) {
+		return BT_HCI_ERR_UNKNOWN_CONN_ID;
+	}
+
+	/*
+	 * EGON TODO: get RSSI information from lll
+	 */
+
+	return BT_HCI_ERR_UNKNOWN_CMD;
+}
+#endif /* CONFIG_BT_CTLR_CONN_RSSI */
+
+/*
+ * the following are only valid for slave configuration
+ */
+#if defined(CONFIG_BT_CTLR_LE_ENC) && defined(CONFIG_BT_PERIPHERAL)
+u8_t ll_start_enc_req_send(u16_t handle, u8_t error_code,
+			    u8_t const *const ltk)
+{
+	struct ull_cp_conn *conn;
+
+	conn = ll_connected_get(handle);
+	if (!conn) {
+		return BT_HCI_ERR_UNKNOWN_CONN_ID;
+	}
+
+	/*
+	 * EGON TODO: add info to the conn-structure
+	 * - refresh
+	 * - no procedure in progress
+	 * - procedure type
+	 * and use that info to decide if the cmd is allowed
+	 * or if we should terminate the connection
+	 * see BT 5.2 Vol. 6 part B chapter 5.1.3
+	 * see also ull_slave.c line 395-439
+	 *
+	 * EGON TODO: the ull_cp_ltx_req* functions should return success/fail status
+	 */
+	if (error_code) {
+		ull_cp_ltk_req_neq_reply(conn);
+		return BT_HCI_ERR_SUCCESS;
+	} else {
+		ull_cp_ltk_req_reply(conn);
+		return BT_HCI_ERR_SUCCESS;
+	}
+
+	return BT_HCI_ERR_UNKNOWN_CMD;
+}
+#endif /* CONFIG_BT_CTLR_LE_ENC && CONFIG_BT_PERIPHERAL */
+
+#if defined(CONFIG_BT_CENTRAL)
+/** @brief Prepare access address as per BT Spec.
+ *
+ * - It shall have no more than six consecutive zeros or ones.
+ * - It shall not be the advertising channel packets' Access Address.
+ * - It shall not be a sequence that differs from the advertising channel
+ *   packets Access Address by only one bit.
+ * - It shall not have all four octets equal.
+ * - It shall have no more than 24 transitions.
+ * - It shall have a minimum of two transitions in the most significant six
+ *   bits.
+ *
+ * LE Coded PHY requirements:
+ * - It shall have at least three ones in the least significant 8 bits.
+ * - It shall have no more than eleven transitions in the least significant 16
+ *   bits.
+ */
+static inline void access_addr_get(u8_t access_addr[])
```

Comment:
There is a helper function for getting AA, which showed up after we did the rebase to master. I decided not to use that function for now, since we need to revisit all functions here in any case before merging back to master